src/libstore/content-address.hh

```diff
@@ -5,44 +5,75 @@
 
 namespace nix {
 
+/**
+ * An enumeration of the ways we can serialize file system objects.
+ */
 enum struct FileIngestionMethod : uint8_t {
+    /**
+     * When we just have a single file, we can directly ingest its
+     * contents
+     */
     Flat = false,
+    /**
+     * For any sort of Nix-support file system object tree, we can
+     * serialize to a ‘Nix Archive’ and then ingest that.
+     */
     Recursive = true
 };
 
+/**
+ * For paths computed by Store::makeTextPath() / Store::addTextToStore
+ */
```

Comment:
I don't think a comment like that brings much (except a risk for drifting out of sync ;) ). What's the reason for this hashing method to exist? Is it just backwards compatibility or does it have a stronger _raison d'être_?

(Same one for `FixedOutputHash` below)

Reply:
Done.